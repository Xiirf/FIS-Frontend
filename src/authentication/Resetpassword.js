import React from 'react';
import { Formik, Field, Form, ErrorMessage } from 'formik';
import * as Yup from 'yup';
import { toast } from 'react-toastify';

import { authenticationService } from '../_services/authentication.service';

class Resetpassword extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            token: null
        };

        // redirect to home if already logged in
        if (authenticationService.currentTokenValue) { 
            this.props.history.push('/');
        }
    }

    componentDidMount() {
        this.setState({ token: this.props.match.params.token })
    }

    success = () => {
      toast.success('La contraseña se ha restablecido con éxito.', {
        position: "top-right",
        autoClose: 10000,
        hideProgressBar: false,
        closeOnClick: true,
        pauseOnHover: true,
        draggable: true
        });
    }

    render() {
        const { token } = this.state;
        return (
            <div>
                <h2>Contraseña olvidada</h2>
                <Formik
                    initialValues={{
                        password: ''
                    }}
                    validationSchema={Yup.object().shape({
                        password: Yup.string().required('Se requiere una contraseña')
                    })}
                    onSubmit={({ password }, { setStatus, setSubmitting }) => {
                        setStatus();
                        authenticationService.resetPassword(token, password)
                            .then(
                                () => {
                                    this.success();
                                    const { from } = this.props.location.state || { from: { pathname: "/signin" } };
                                    this.props.history.push(from);
                                },
                                error => {
                                    setSubmitting(false);
                                    setStatus(error);
                                }
                            );
                    }}
                    render={({ errors, status, touched, isSubmitting }) => (
                        <Form>
                            <div className="form-group">
                                <label htmlFor="password">Contraseña</label>
                                <Field name="password" type="password" className={'form-control' + (errors.password && touched.password ? ' is-invalid' : '')} />
                                <ErrorMessage name="password" component="div" className="invalid-feedback" />
                            </div>
                            <div className="form-group">
                                <button type="submit" className="btn btn-primary" disabled={isSubmitting}>Enviar</button>
                                {isSubmitting &&
                                    <img src="data:image/gif;base64,R0lGODlhEAAQAPIAAP///wAAAMLCwkJCQgAAAGJiYoKCgpKSkiH/C05FVFNDQVBFMi4wAwEAAAAh/hpDcmVhdGVkIHdpdGggYWpheGxvYWQuaW5mbwAh+QQJCgAAACwAAAAAEAAQAAADMwi63P4wyklrE2MIOggZnAdOmGYJRbExwroUmcG2LmDEwnHQLVsYOd2mBzkYDAdKa+dIAAAh+QQJCgAAACwAAAAAEAAQAAADNAi63P5OjCEgG4QMu7DmikRxQlFUYDEZIGBMRVsaqHwctXXf7WEYB4Ag1xjihkMZsiUkKhIAIfkECQoAAAAsAAAAABAAEAAAAzYIujIjK8pByJDMlFYvBoVjHA70GU7xSUJhmKtwHPAKzLO9HMaoKwJZ7Rf8AYPDDzKpZBqfvwQAIfkECQoAAAAsAAAAABAAEAAAAzMIumIlK8oyhpHsnFZfhYumCYUhDAQxRIdhHBGqRoKw0R8DYlJd8z0fMDgsGo/IpHI5TAAAIfkECQoAAAAsAAAAABAAEAAAAzIIunInK0rnZBTwGPNMgQwmdsNgXGJUlIWEuR5oWUIpz8pAEAMe6TwfwyYsGo/IpFKSAAAh+QQJCgAAACwAAAAAEAAQAAADMwi6IMKQORfjdOe82p4wGccc4CEuQradylesojEMBgsUc2G7sDX3lQGBMLAJibufbSlKAAAh+QQJCgAAACwAAAAAEAAQAAADMgi63P7wCRHZnFVdmgHu2nFwlWCI3WGc3TSWhUFGxTAUkGCbtgENBMJAEJsxgMLWzpEAACH5BAkKAAAALAAAAAAQABAAAAMyCLrc/jDKSatlQtScKdceCAjDII7HcQ4EMTCpyrCuUBjCYRgHVtqlAiB1YhiCnlsRkAAAOwAAAAAAAAAAAA==" />
                                }
                            </div>
                            {status &&
                                <div className={'alert alert-danger'}>{status}</div>
                            }
                        </Form>
                    )}
                />
            </div>
        )
    }
}

export default Resetpassword;