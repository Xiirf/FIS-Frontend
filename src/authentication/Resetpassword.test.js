import React from 'react';
import Resetpassword from './Resetpassword';
import { render, unmountComponentAtNode } from 'react-dom';
import { create } from "react-test-renderer";

let container = null;
beforeEach(() => {
  // met en place un élément DOM comme cible de rendu
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // nettoie en sortie de test
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('renders without crashing', () => {
  render(<Resetpassword match={{params: {token: 12345}}} />, container);
});

it("matches the snapshot", () => {
    const resetpassword = create(<Resetpassword match={{params: {token: 12345}}} />);
    expect(resetpassword.toJSON()).toMatchSnapshot();
});