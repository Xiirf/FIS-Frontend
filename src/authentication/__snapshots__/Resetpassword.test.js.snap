// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<div>
  <h2>
    Contraseña olvidada
  </h2>
  <form
    action="#"
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <label
        htmlFor="password"
      >
        Contraseña
      </label>
      <input
        className="form-control"
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        type="password"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <button
        className="btn btn-primary"
        disabled={false}
        type="submit"
      >
        Enviar
      </button>
    </div>
  </form>
</div>
`;
